integrations/pagerduty/clients/pagerduty.py

```diff
@@ -67,7 +67,13 @@ async def paginate_request_to_pager_duty(self, data_key: str) -> list[Any]:
                     )
                     response.raise_for_status()
                     data = response.json()
-                    all_data.extend(data[data_key])
+
+                    # Fetch and update on-call user information for services
+                    if data_key == "services":
+                        service_data = await self.update_oncall_users(data, data_key)
+                        all_data.extend(service_data)
+                    else:
+                        all_data.extend(data[data_key])
```

Comment:
Why are you doing this?
This is being done on main.py

Reply:
My personal repo may have been out of sync with the current. Will update it accordingly